gdal_subsetter/transform.py

```diff
@@ -447,109 +450,86 @@ def get_shapefile(self, subsetshape, dstdir):
         normalized_layerid = layerid.replace("/", "_")
         dstfile = os.path.join(dstdir, f"{normalized_layerid}__reprojected.tif")
-        fmt = self.message.format
-        if fmt.scaleSize:
-            xres, yres = fmt.process("scaleSize").x, fmt.process("scaleSize").y
-        else:
-            xres, yres = None, None
-        if fmt.scaleExtent:
-            box = [
-                fmt.process("scaleExtent").x.min,
-                fmt.process("scaleExtent").y.min,
-                fmt.process("scaleExtent").x.max,
-                fmt.process("scaleExtent").y.max,
-            ]
-        else:
-            box = None
-
-        def _regrid(
-            infile,
-            outfile,
-            resampling_mode="bilinear",
-            ref_crs=None,
-            ref_box=None,
-            ref_xres=None,
-            ref_yres=None,
-        ):
-            command = ["gdalwarp", "-of", "GTiff", "-overwrite", "-r", resampling_mode]
-
-            if ref_crs:  # proj, box, xres/yres
-                command.extend(["-t_srs", ref_crs])
-                # command.extend([ '-t_srs', "'{ref_crs}'".format(ref_crs=ref_crs)])
-            if ref_box:
-                box = [str(x) for x in ref_box]
-                command.extend(
-                    ["-te", box[0], box[1], box[2], box[3], "-te_srs", ref_crs]
-                )
-                # command.extend(['-te', box[0], box[1], box[2], box[3],
-                #                 '-te_srs', "'{ref_crs`}'".format(ref_crs=ref_crs)])
-            if ref_xres and ref_yres:
-                command.extend(["-tr", str(ref_xres), str(ref_yres)])
-
-            command.extend([infile, outfile])
-            self.execute_gdal_command(*command)
-            return outfile
-
-        return _regrid(
-            srcfile,
-            dstfile,
-            resampling_mode=resample_method,
-            ref_crs=crs,
-            ref_box=box,
-            ref_xres=xres,
-            ref_yres=yres,
-        )
+
+        output_crs = rgetattr(self.message, "format.crs", None)
+
+        if output_crs is None:
+            # No reprojection requested, return path to input file
+            return srcfile
+
+        warp_options_dict = {
+            "dstSRS": output_crs,
+            "format": "GTiff",
+            "outputBoundsSRS": output_crs,
+            "resampleAlg": get_resample_algorithm(self.message),
+        }
+
+        # Extract requested x and y resolutions for output
+        output_xres = rgetattr(self.message, "format.scaleSize.x", None)
+        output_yres = rgetattr(self.message, "format.scaleSize.y", None)
+
+        if output_xres is not None and output_yres is not None:
+            warp_options_dict["xRes"] = output_xres
+            warp_options_dict["yRes"] = output_yres
+
+        # Extract requested bounding box
+        output_x_min = rgetattr(self.message, "format.scaleExtent.x.min", None)
+        output_x_max = rgetattr(self.message, "format,scaleExtent.x.max", None)
```

Comment:
```suggestion
        output_x_max = rgetattr(self.message, "format.scaleExtent.x.max", None)
```

Reply:
_Great_ catch! I think also, this speaks to my plan to move some of these methods to be functions in a separate module and throw a bunch of tests around them. I still think that should happen after this PR, though.

Reply:
(Also more regression tests)

Reply:
Fixed in [62f49ef](https://github.com/nasa/harmony-gdal-adapter/pull/42/commits/62f49eff7e88cb5183e6e44a34fefa0a0bcff70b).